src/host/usbh.c

```diff
@@ -557,6 +563,14 @@ void tuh_task_ext(uint32_t timeout_ms, bool in_isr) {
         TU_LOG_USBH("[%u:%u:%u] USBH DEVICE REMOVED\r\n", event.rhport, event.connection.hub_addr, event.connection.hub_port);
         process_removing_device(event.rhport, event.connection.hub_addr, event.connection.hub_port);
 
+        if ((event.rhport == _dev0.rhport) && (event.connection.hub_addr == _dev0.hub_addr) &&
+            (event.connection.hub_port == _dev0.hub_port)) {
+          _dev0.enumerating = 0;
+          hcd_device_close(_dev0.rhport, _dev0.hub_addr);
```

Comment:
should include `_dev0.enumerating == 1` since other value has obsolete (for new device) after enumerated successfully. The isr blindly set the flag, but we need to check since we do more than that here. Also `hcd_device_close(_dev0.rhport, 0)` instead of using hub_addr

Reply:
I haven't tested, is `hcd_device_close(_dev0.rhport, 0)` works with device under a hub ?

Reply:
It should, the 2nd argument is the device addres that is removed. Otherwise, it will try to remove the hub that attached the rhport.

Reply:
I think there is a loophole with early return. If set address is done `_dev0.enumerating` is still true but `_usbh_devices` also needs to be cleaned.

Reply:
ah you are right, putting enumerating in dev0 is confusing. Let me revise and fix this.